lms/urls.py

```diff
@@ -714,36 +732,11 @@
             ),
             include('lms.djangoapps.discussion.django_comment_client.urls')
         ),
-        url(
-            r'^notification_prefs/enable/',
-            notification_prefs_views.ajax_enable
-        ),
-        url(
-            r'^notification_prefs/disable/',
-            notification_prefs_views.ajax_disable
-        ),
-        url(
-            r'^notification_prefs/status/',
-            notification_prefs_views.ajax_status
-        ),
-        url(
-            r'^notification_prefs/unsubscribe/(?P<token>[a-zA-Z0-9-_=]+)/',
-            notification_prefs_views.set_subscription,
-            {
-                'subscribe': False,
-            },
-            name='unsubscribe_forum_update',
-        ),
-        url(
-            r'^notification_prefs/resubscribe/(?P<token>[a-zA-Z0-9-_=]+)/',
-            notification_prefs_views.set_subscription,
-            {
-                'subscribe': True,
-            },
-            name='resubscribe_forum_update',
-        ),
     ]
 
+if settings.FEATURES.get('ENABLE_FORUM_DAILY_DIGEST'):
```

Comment:
If you're switching over to use Django settings, please put in default value and annotations in the LMS `common.py` file.

Reply:
Thanks, I am going to do that. If you could add some context why this setting is required after we have added the flag in edx-internal, it would be helpful. thanks.

Reply:
It's mostly so that we can have a central list of the toggles in platform, with default values and standardized documentation (the `.. togle_name: ...` stuff).

Reply:
@awaisdar001 @ormsbee: I need to remove or refactor any waffle flags using `            flag_undefined_default=True`.  I'm confused about this PR.

This merged PR has `ENABLE_FORUM_DAILY_DIGEST` as both a Django Setting **and** a waffle flag. Was the waffle flag supposed to have been removed, but never was?  Some notes about the waffle flag:
1. Although the flag is defined with `flag_undefined_default=True`, the flag is turned **off** in Production and Edge.
2. The function `is_forum_daily_digest_enabled()`, which encapsulates the waffle flag, is used in several places.
3. The Django Setting of the same name is set to True in Stage and Production, but not in Edge.

I need help understanding if the waffle flag can be removed and if its usage should be replaced by the setting, even though that seems like it will change its value.  Or, other?  Thank you.

Reply:
@awaisdar001: Dave mentioned he doesn't know the background, so I'll need your help determining how we can retire the waffle flag.
1. Can we simply remove the waffle flag and replace uses of `is_forum_daily_digest_enabled` with a check for the Django setting?
2. If we do this, I think the value will go from False to True in Production, because the waffle flag was False, but the Django setting was True.  Is this ok?  If not, what else is needed?
Thank you.

Reply:
@robrap You are right, there are two identifiers (1. Django Settings, 2. WaffleSwitch).
1. Django settings are used to control whether or not to include forum digest URLS into Django Url confs. Waffle switch could be used at this stage b/c models access fails at Django initialization. 
2. Waffle Switch is used to control UI. Whether or not we need to render the Forum Diest UI.

WaffleSettings was added in Prod with `flag_undefined_default=True` and was disabled latter on the specific date.

Reply:
Thanks @awaisdar001.
1. When you say `WaffleSwitch`, `Waffle Switch`, and `WaffleSettings`, did you always mean `WaffleFlag`?
2. What is the long-term intent of the WaffleFlag? Is it ready to be deleted and always set to False?  Or, is it meant to be a permanent setting for the community?  Note that in Juniper, the community has this flag set to True by default (as written).

Basically, can I remove the Waffle Flag?  If not, can you help me understand how to document its long term use so I know how to properly update it?

Thank you.

Reply:
Also, could I simply change the WaffleFlag default to False, which would affect Open edX, devstack, and sandboxes? If the flag can also be removed, I may do whatever is simpler.

Reply:
> Thanks @awaisdar001.
> 
> 1. When you say `WaffleSwitch`, `Waffle Switch`, and `WaffleSettings`, did you always mean `WaffleFlag`?
> 2. What is the long-term intent of the WaffleFlag? Is it ready to be deleted and always set to False?  Or, is it meant to be a permanent setting for the community?  Note that in Juniper, the community has this flag set to True by default (as written).
> 
> Basically, can I remove the Waffle Flag? If not, can you help me understand how to document its long term use so I know how to properly update it?
> 
> Thank you.


1. Yes, sorry if it caused confusion, I used the word interchangeably. 
2. I think the conversation on the ticket https://openedx.atlassian.net/browse/PROD-1342?focusedCommentId=467317 explains it. 


@robrap If you are to remove the flag, please make sure the AC from the ticket still remains the same. 

3. I think it's ok to have a Django Setting for both Notifier URLs and UI and it is safe to remove the `WaffleFlag` until it satisfies the AC. 


4. Thanks for flagging that DjangoSettings is disabled and endpoints are still exposed. The PR was reverted https://github.com/edx/edx-internal/pull/1755 sometime after it was merged. I think we need to "Revert the "Reverted PR".

Reply:
Great. Thanks. I think I have what I need to resolve. 

The plan:
1. Update the setting values to be the same as waffle, and matching AC.
2. Retire the waffle flag and replace it with code that uses the setting.